git-commit-signing/README.md

```diff
@@ -0,0 +1,23 @@
+---
+display_name: Git commit signing
+description: Configures Git to sign commits using your Coder SSH key
+icon: ../.icons/git.svg
+maintainer_github: phorcys420
+verified: false
+tags: [helper, git]
+---
+
+# git-commit-signing
+
+This module downloads your SSH key from Coder and uses it to sign commits with Git.
+It requires `jq` to be installed inside your workspace.
+
+This is not recommended if your workspace can be accessed by other/unwanted people, in the case an administrator account on your Coder account gets breached, the attacker could gain access to your workspace and sign commits on your behalf (since the key is stored in the workspace).
+If your Coder account gets breached, the SSH key could also be used on your behalf.
```

Comment:
I think we could simplify this disclaimer somewhat as I don't think this module actually increases risk by much. The only difference to not using this module is that the SSH key is stored on disk, i.e. your home volume.

If an admin gained access to your workspace, they could still issue the `curl` command used here to get your key. Likewise, if your Coder account is breached, the key can simply be pulled from the server.

Reply:
i see what you mean but what i meant is the risk of adding that ssh key as a signing key means that you kinda compromise signing

Reply:
Ah, I see what you mean. Perhaps changing the message slightly would make sense then?

> Please observe that using the SSH key that's part of your Coder account for commit signing, means that in the event of a breach of your Coder account, or malicious admin, they could perform commit signing pretending to be you.

Apologies if the wording is bad, I mainly wanted to convey my thought, feel free to use it, a part of it, or not at all. 😄

Reply:
No, I actually thing the wording is pretty good and not too long to read like mine.

Reply:
3e9e6aad0b0ad9c2569419f3a29ed541cfcf0506

Reply:
3e9e6aad0b0ad9c2569419f3a29ed541cfcf0506

Reply:
326fafad027dc4462292a5d3ec84a869d66aa9a3

Reply:
bf7f9d04163e9c085a7c57306e4198aa84019fb0